### ISSUE TITLE
Parser chokes on given file

### ISSUE THREAD

**Issue by nicStuff** [arashm/PoParser#1]
Hi, I tried to use the parser on a .po file but it chokes giving the following message:

```
~/.rvm/gems/ruby-2.2.3/gems/PoParser-1.0.0/lib/poparser/transformer.rb:10:in `transform': undefined method `each' for "":String (NoMethodError)
    from ~/.rvm/gems/ruby-2.2.3/gems/PoParser-1.0.0/lib/poparser/tokenizer.rb:31:in `parse_block'
    from ~/.rvm/gems/ruby-2.2.3/gems/PoParser-1.0.0/lib/poparser/tokenizer.rb:15:in `block (2 levels) in extract_entries'
    from ~/.rvm/gems/ruby-2.2.3/gems/PoParser-1.0.0/lib/poparser/tokenizer.rb:13:in `each_line'
    from ~/.rvm/gems/ruby-2.2.3/gems/PoParser-1.0.0/lib/poparser/tokenizer.rb:13:in `block in extract_entries'
    from ~/.rvm/gems/ruby-2.2.3/gems/PoParser-1.0.0/lib/poparser/tokenizer.rb:12:in `open'
    from ~/.rvm/gems/ruby-2.2.3/gems/PoParser-1.0.0/lib/poparser/tokenizer.rb:12:in `extract_entries'
    from ~/.rvm/gems/ruby-2.2.3/gems/PoParser-1.0.0/lib/poparser.rb:19:in `parse'
    from test_script.rb:350:in `block in test_method'
    from test_script.rb:342:in `foreach'
    from test_script.rb:342:in `test_method'
    from test_script.rb:403:in `<main>'
```

Unfortunately I cannot share the .po file since it's of my company: so for the moment this issue is more a reminder, if it's a real problem. Tried on another .po file and runs fine.

**Comment by arashm** [comment]
If you could make a partial of that file (with different strings) which generates the same error and share it, I could take a look.

**Comment by nicStuff** [comment]
I thoght about doing it but I'm very short in time, sorry. Personally I
don't need the fix anymore so I won't take a look. I think this can be
useful if someone will have the same error. Thank you for your library
Il 18/set/2015 18:19, "Arash Mousavi" notifications@github.com ha scritto:

**Comment by roland9** [comment]
having the same issue - at least in my case, it's happening when the `app.po` file starts with an empty line :)  this means `block` is still `''` when it was initialised 

```

# SOME DESCRIPTIVE TITLE.
# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
# This file is distributed under the same license as the app package.
# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
#
msgid ""
msgstr ""
"Project-Id-Version: app 1.0.0\n"
```

removing that first empty line avoids this issue.

suggested fix #2

**Comment by roland9** [comment]
remark: I'm not sure though if that would still be considered a valid `po file`??  (when it starts with an empty line?)
anyway, it should give a better error message - or handle it gracefully

**Comment by dfherr** [comment]
this should be fixed in version 3.0.0
